### ISSUE TITLE
Support loading multiple soundfonts

### ISSUE THREAD

**Issue by GranMinigun** [dosbox-staging/dosbox-staging#1080]
Fluid R3 soundfont comes in two parts: main one with General MIDI set (`FluidR3_GM.sf2`), and General Standard subset (`FluidR3_GS.sf2`). Currently there's no way to load them both in dosbox-staging.

Proposal: allow to specify comma-separated list as a first argument to `soundfont` setting (to preserve gain value as a second argument). Load soundfonts in the order specified, with later soundfonts overriding previous ones:
```
soundfont = FluidR3_GM.sf2,FluidR3_GS.sf2 50
```

The catch is, while reasonable number of games supported Sound Canvas, I don't know any that actually used GS instruments.

**Comment by kcgen** [comment]
This should be doable. 

In the meantime: https://musical-artifacts.com/artifacts/1229

**Comment by dreamer** [comment]
We were asked several times about this already… Is there a single piece of DOS software that uses GS MIDI? ;)

I don't think we should do this.

**Comment by kcgen** [comment]
@dreamer noted:
> I tried cppcheck in the past (before clang-static-analyzer) - it was very bad back then. Lots of false-positives, meaningless issues, and things contradicting established style (stuff that shouldn't be part of analysis). Maybe it improved, but I don't trust that tool.

**Comment by kcgen** [comment]
Suggest we close this for now, given we've already got our hands full with Coverity, PVS Studio ( the latter keeps improving too), checked STL, Clang's checkers, plus the runtime sanitizers - this is likely sufficient to have us covered.  

 If we ever exhaust all of those, then maybe we won't be writing C++ anymore, either. :-)

**Comment by kcgen** [comment]
Even though we've closed this, tickets like this show you're looking out for the project and trying to help make it better.  Thank you @LowLevelMahn !

**Comment by kcgen** [comment]
Accidentally closed.

**Comment by kcgen** [comment]
@GranMinigun - I recall us trying to figure out if any DOS game supposed the extended GS set; and I think we came up empty? So perhaps this is outside the scope of DOS gaming.. even though FluidSynth technically can support it.

**Comment by kcgen** [comment]
@GranMinigun , I'm happy to commit to this work if you can post a comparison showing improved MIDI music using external FluidSynth with layered GM+GS soundfonts versus the embedded-FluidSynth with the GM-only soundfont.

**Comment by dragonwolf83** [comment]
> Is there a single piece of DOS software that uses GS MIDI? ;)

Yes. I found these two links that have a list of some games that used GS. 
https://www.vogonswiki.com/index.php/List_of_games_by_MIDI_standard
https://nerdlypleasures.blogspot.com/2012/04/unique-pc-hardware-game-support.html

I also know that Tom Lewandowski from [Quest Studios](http://www.midimusicadventures.com/queststudios/) is an expert on MIDI with Sierra Games. He did alot of digital recordings of the games to get their original sound in best/as-intended quality. His site has midi files for the games intended for MT-32, GS, or GM to match what the game supported. He would be the best person to ask about GS and MIDI in general.

**Comment by kcgen** [comment]
@GranMinigun - are we OK to close this? given that GS under DOS expects a Roland's SC-55 instrument set, and therefore any SC-55-compatible soundfont should provide those same extended instruments as the SC-55.

@HW-Scanno mentioned that  "generalized" GS (non-SC-55-compatible) instruments are undefined, and therefore would produce unpredictable results under DOS games, which I think your tests bore out.

(Ref: https://github.com/dosbox-staging/dosbox-staging/pull/1125).
